crates/telio-starcast/src/nat.rs

```diff
@@ -0,0 +1,648 @@
+use crate::utils::MutableIpPacket;
+use pnet_packet::{
+    ip::IpNextHeaderProtocols,
+    ipv4::{self, MutableIpv4Packet},
+    ipv6::MutableIpv6Packet,
+    udp::{self, MutableUdpPacket},
+};
+use std::{
+    hash::{DefaultHasher, Hash, Hasher},
+    net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr},
+    time::Duration,
+};
+use telio_utils::{telio_log_warn, LruCache};
+
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    /// The packet is too short to be a valid IP packet
+    PacketTooShort,
+    /// The packet has an unexpected IP version
+    UnexpectedIpVersion,
+    /// The packet has an unexpected transport protocol
+    UnexpectedTransportProtocol,
+    /// There is no mapping for the given port
+    MappingNotFound,
+}
+
+pub trait Nat {
```

Comment:
Lets add a rustdoc comment explaining that outgoing direction is transport socket to other peer.
And packet's are incomming from tunnel / virtual peer

Reply:
It's actually the other way around ;) But I'll comment it to make it clear